src/components/SearchBar/SearchBar.vue

```diff
@@ -0,0 +1,154 @@
+<template>
+  <div ref="searchBar">
+    <text-input
+      id="searchBar"
+      v-model="searchTerm"
+      class="w-max bg-white-300 h-12"
+      grey
+    >
+      <template #iconLeft>
+        <search class="w-4 h-6" />
+      </template>
+      <template #iconRight>
+        <button
+          :role="disabled ? 'button' : null"
```

Comment:
You don't need this role on a semantic HTML button element. It's already a button so this could actually introduce issues in screenreaders.